snapshotter/app/service.go

```diff
@@ -63,17 +63,18 @@ func Run(config config.Config) error {
 	if config.Snapshotter.Metrics.Enable {
 		sdHost := config.Snapshotter.Metrics.Host
 		sdPort := config.Snapshotter.Metrics.ServiceDiscoveryPort
-		serviceDiscovery := discovery.NewServiceDiscovery(sdHost, sdPort, cache)
-		monitor, err := initMetricsProxyMonitor(config.Snapshotter.Metrics.PortRange)
+		serviceDiscovery = discovery.NewServiceDiscovery(sdHost, sdPort, cache)
+		mtr, err := initMetricsProxyMonitor(config.Snapshotter.Metrics.PortRange)
```

Comment:
What's going on here? Is there some issue with aliasing?

Reply:
yes, it's issue with aliasing/scoping. `monitor, err := ...`, `serviceDiscovery := ...` would create new objects, but the variables initalized outside this scope:
```
var (
		monitor          *metrics.Monitor
		serviceDiscovery *discovery.ServiceDiscovery
)
```

would not actually be set properly; `monitor` and `serviceDiscovery` would be `nil`

Reply:
I updated to remove `mtr` to be more consistent / readable.